packages/api/src/brevo-config/brevo-config.resolver.ts

```diff
@@ -0,0 +1,97 @@
+import { AffectedEntity, RequiredPermission, validateNotModified } from "@comet/cms-api";
+import { EntityManager, EntityRepository, wrap } from "@mikro-orm/core";
+import { InjectRepository } from "@mikro-orm/nestjs";
+import { Type } from "@nestjs/common";
+import { Args, ID, Mutation, Query, Resolver } from "@nestjs/graphql";
+
+import { BrevoApiSenderService } from "../brevo-api/brevo-api-sender.service";
+import { BrevoApiSender } from "../brevo-api/dto/brevo-api-sender";
+import { EmailCampaignScopeInterface } from "../types";
+import { DynamicDtoValidationPipe } from "../validation/dynamic-dto-validation.pipe";
+import { BrevoConfigInput, BrevoConfigUpdateInput } from "./dto/brevo-config.input";
+import { BrevoConfigInterface } from "./entities/brevo-config-entity.factory";
+
+export function createBrevoConfigResolver({
+    Scope,
+    BrevoConfig,
+}: {
+    Scope: Type<EmailCampaignScopeInterface>;
+    BrevoConfig: Type<BrevoConfigInterface>;
+}): Type<unknown> {
+    @Resolver(() => BrevoConfig)
+    @RequiredPermission(["brevo-newsletter-config"])
+    class BrevoConfigResolver {
+        constructor(
+            private readonly entityManager: EntityManager,
+            private readonly brevoSenderApiService: BrevoApiSenderService,
+            @InjectRepository(BrevoConfig) private readonly repository: EntityRepository<BrevoConfigInterface>,
+        ) {}
+
+        private async isValidSender({ email, name }: { email: string; name: string }): Promise<boolean> {
+            const senders = await this.brevoSenderApiService.getSenders(Scope);
+
+            if (senders && senders.some((sender) => sender.email === email && sender.name === name)) {
+                return true;
+            }
+
+            return false;
+        }
+
+        @RequiredPermission(["brevo-newsletter-config"], { skipScopeCheck: true })
```

Comment:
Why is the skipScope check necessary here?

Reply:
Corrected: c01b7a05fb5f2cf03f09c88654f6838281b0f472